core/api/src/main/java/org/eclipse/sensinact/core/whiteboard/WhiteboardGetSet.java

```diff
@@ -0,0 +1,68 @@
+/*********************************************************************
+* Copyright (c) 2024 Kentyou.
+*
+* This program and the accompanying materials are made
+* available under the terms of the Eclipse Public License 2.0
+* which is available at https://www.eclipse.org/legal/epl-2.0/
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*   Thomas Calmant (Kentyou) - initial implementation
+**********************************************************************/
+package org.eclipse.sensinact.core.whiteboard;
+
+import java.time.Duration;
+
+import org.eclipse.sensinact.core.twin.TimedValue;
+import org.osgi.util.promise.Promise;
+import org.osgi.util.promise.PromiseFactory;
+
+/**
+ * Whiteboard service to handle resource get/set operations
+ *
+ * @param <T> Resource type
+ */
+public interface WhiteboardGetSet<T> {
+
+    /**
+     * Class of the resource
+     */
+    Class<T> getResourceType();
+
+    /**
+     * Cache duration
+     */
+    Duration getCacheDuration();
+
+    /**
+     * Whiteboard is called to get the resource value as a promise
+     *
+     * @param promiseFactory  Promise factory
+     * @param modelPackageUri Package URI of the provider model
+     * @param model           Provider model name
+     * @param provider        Provider name
+     * @param service         Service name
+     * @param resource        Resource name
+     * @param cachedValue     Previously cached value
+     * @return A promise, created with the promise factory
+     */
+    Promise<TimedValue<T>> pullValue(PromiseFactory promiseFactory, String modelPackageUri, String model,
+            String provider, String service, String resource, TimedValue<T> cachedValue);
+
+    /**
+     * Whiteboard is called to set the resource value
+     *
+     * @param promiseFactory  Promise factory
+     * @param modelPackageUri Package URI of the provider model
+     * @param model           Provider model name
+     * @param provider        Provider name
+     * @param service         Service name
+     * @param resource        Resource name
+     * @param cachedValue     Previously cached value
+     * @param newValue        New value
+     * @return A promise, created with the promise factory
+     */
+    Promise<TimedValue<T>> pushValue(PromiseFactory promiseFactory, String modelPackageUri, String model,
```

Comment:
The annotated version allows you to request the target type as a parameter, which I think that we should include here.

Reply:
I'm not sure it's necessary here, as the type comes from the class implementing pullValue itself to define the resource.
In the annotated case, it's useful as the type might not be easily accessible otherwise.